### DISCUSSION TITLE
Custom docBlock use internal router

### DISCUSSION THREAD

**Discussion by Fernker** [storybookjs/storybook#25762] [Help]
### Summary

I am trying to create a custom docBlock to generate some information for a story.  The data that is passed in includes information about linked stories.  I'm able to generate a URL and have it correctly go to that page.  But because I'm using a regular `anchor` tag it does a full refresh of the page to go to the link.  I'd prefer to have it route within the single-page as it's a lot faster.

Right now this is the code I have, it again works but I've been digging through everything from `useStorybookApi` to `useNavigate` and I can't seem to figure out how to generate a link that uses the internal Storybook router, only the browser one, which is slow since it has to reload Storybook. 

LinkedStories is an import of another set of stories like this:

`import * as FooStories from '../another-component/another-component.stories';`

```
import React from 'react';
import { useOf } from '@storybook/blocks';

export const LinkedStories = ({ of }) => {
  const resolvedMeta = useOf(of, ['meta']);

  const linkedStories = resolvedMeta.preparedMeta.parameters.linkedStories;

  return React.createElement('a', {
    href: `/?path=/docs/${linkedStories.default.title
          .toLowerCase()
          .replace(/ /g, '-')
          .replace(/\//g, '-')}--guide`,
          children: 'Linked Stories'
    });
  }
```

tl;dr; Is there anyway to use the internal router so that it uses the single-page application router when in a custom docBlock?
